Source/WebKit/NetworkProcess/Downloads/cocoa/DownloadCocoa.mm

```diff
@@ -78,15 +79,18 @@
 
 void Download::platformDestroyDownload()
 {
+#if HAVE(MODERN_DOWNLOADPROGRESS)
+    m_bookmarkURL = nil;
+    [m_progress unpublish];
+    [m_progress cancel];
```

Comment:
I think we probably shouldn't call cancel here unless it's actually needed for something.  It should've already been cancelled by this point.